### PR TITLE
fix: Compression config map serialization

### PR DESCRIPTION
**Pull request by jorgebay**
Solves a deserialization issue in `sinks::util::buffer::compression` when the provided value was a map.

Fixes: #12919

### CODE REVIEW

File: src/sinks/util/buffer/compression.rs [hunk 1]
```diff
@@ -305,4 +305,21 @@ mod test {
             assert_eq!(error.to_string().as_str(), *result);
         }
     }
+
+    #[test]
+    fn from_and_to_value() {
```
Comment (jorgebay) [inline src/sinks/util/buffer/compression.rs:310]: This test fails without the fix :)

**Comment by jorgebay** [comment]
I've checked the rest of `visit_map()` implementations and those look good 👍

**Comment by spencergilbert** [comment]
`check-version` should pass if you merge `master` into your branch 👍

**Review by spencergilbert — APPROVED** [review]

**Comment by jorgebay** [comment]
Thanks, I've rebased it against `master`.

**Review by neuronull — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**